src/overlays/actors/ovl_En_Mt_tag/z_en_mt_tag.c

```diff
@@ -35,42 +40,474 @@ const ActorInit En_Mt_tag_InitVars = {
+    Audio_QueueSeqCmd(0x101400FF);
+    this->actionFunc = EnMttag_PotentiallyRestartRace;
+}
+
+/**
+ * Displays the text from the Goron Elder's child which tells the player that
+ * they probably can't win the race.
+ */
+void EnMttag_ShowCantWinMessage(EnMttag* this, GlobalContext* globalCtx) {
+    func_801518B0(globalCtx, 0xE97, NULL); // You can't win now...
+    func_800B7298(globalCtx, &this->actor, 7);
+    this->actionFunc = EnMttag_HandleCantWinChoice;
+}
+
+/**
+ * Shows the cutscene that pans over the race track and shows all five race entrants.
+ */
+void EnMttag_ShowIntroCutscene(EnMttag* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        this->actionFunc = EnMttag_WaitForIntroCutsceneToEnd;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
+
+/**
+ * When the intro cutscene concludes, this sets the weekEventReg to prevent it
+ * from showing again and starts the race.
+ */
+void EnMttag_WaitForIntroCutsceneToEnd(EnMttag* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCurrentIndex() != this->actor.cutscene) {
+        gSaveContext.weekEventReg[12] |= 2;
+        this->actionFunc = EnMttag_RaceStart;
+    }
+}
+
+/**
+ * Handles the race from when the Gorons are first lined up at the
+ * starting block to when the countdown finishes.
+ */
+void EnMttag_RaceStart(EnMttag* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s32 playerCheatStatus;
+
+    if (this->raceInitialized == true) {
+        playerCheatStatus = EnMttag_CheckPlayerCheatStatus(&player->actor.world.pos);
+        if (playerCheatStatus != GORON_RACE_CHEAT_NO_CHEATING) {
+            if (playerCheatStatus == GORON_RACE_CHEAT_FALSE_START) {
+                this->shouldRestartRace = true;
+            } else {
+                this->shouldRestartRace = false;
+            }
+
+            EnMttag_ShowFalseStartMessage(this, globalCtx);
+            gSaveContext.eventInf[1] |= 8;
+        } else {
+            if (DECR(this->timer) == 60) {
+                func_8010E9F0(4, 0);
+                globalCtx->interfaceCtx.unk_280 = 1;
+                Audio_QueueSeqCmd(NA_BGM_GORON_RACE | 0x8000);
+                globalCtx->envCtx.unk_E4 = 0xFE;
+                player->stateFlags1 &= ~0x20;
+            } else if ((this->timer < 60) && (globalCtx->interfaceCtx.unk_280 == 8)) {
+                this->timer = 0;
+                gSaveContext.eventInf[1] |= 1;
+                this->actionFunc = EnMttag_Race;
+            }
+        }
+    } else {
+        if (EnMttag_AreFourRaceGoronsPresent(this, globalCtx)) {
+            this->raceInitialized = true;
+        }
+    }
+}
+
+/**
+ * Returns true if any Race Goron is in the finish line.
+ */
+s32 EnMttag_IsAnyRaceGoronInFinishLine(EnMttag* this) {
+    s32 isAnyRaceGoronFinished = false;
```

Comment:
I don't like the wording much here, I think it should be either "HasAnyRaceGoronFinished" or "IsAnyRaceGoronOverFinishLine"